admin/app/components/solidus_admin/ui/forms/text_field/component.rb

```diff
@@ -0,0 +1,155 @@
+# frozen_string_literal: true
+
+class SolidusAdmin::UI::Forms::TextField::Component < SolidusAdmin::BaseComponent
+  SIZES = {
+    s: 'leading-4 body-small',
+    m: 'leading-6 body-small',
+    l: 'leading-9 body-text'
+  }.freeze
+
+  TYPES = {
+    color: :color_field,
+    date: :date_field,
+    datetime: :datetime_field,
+    email: :email_field,
+    month: :month_field,
+    number: :number_field,
+    password: :password_field,
+    phone: :phone_field,
+    range: :range_field,
+    search: :search_field,
+    text: :text_field,
+    time: :time_field,
+    url: :url_field,
+    week: :week_field
+  }.freeze
+
+  # @param field [Symbol] the name of the field. Usually a model attribute.
+  # @param form [ActionView::Helpers::FormBuilder] the form builder instance.
+  # @param type [Symbol] the type of the field. Defaults to `:text`.
+  # @param size [Symbol] the size of the field: `:s`, `:m` or `:l`.
+  # @param hint [String, null] helper text to display below the field.
+  # @param errors [Hash, nil] a Hash of errors for the field. If `nil` and the
+  #   form is bound to a model instance, the component will automatically fetch
+  #   the errors from the model.
+  # @param attributes [Hash] additional HTML attributes to add to the field.
+  # @raise [ArgumentError] when the form builder is not bound to a model
+  #  instance and no `errors` Hash is passed to the component.
+  def initialize(field:, form:, type: :text, size: :m, hint: nil, errors: nil, **attributes)
+    @field = field
+    @form = form
+    @type = type
+    @size = size
+    @hint = hint
+    @type = type
+    @attributes = attributes
+    @errors = errors || @form.object&.errors || raise(ArgumentError, <<~MSG
+      When the form builder is not bound to a model instance, you must pass an
+      errors Hash (`field_name: [errors]`) to the component.
+    MSG
+    )
+  end
+
+  def call
+    tag.div do
+      label_tag + field_tag + hint_tag + error_tag
+    end
+  end
+
+  def label_tag
+    @form.label(@field, class: "block mb-0.5 body-tiny-bold")
+  end
+
+  def field_tag
+    @form.send(
+      field_helper,
+      @field,
+      class: "
```

Comment:
Would be nice to have the classes as a method, it should be easier to read and to overwrite if someone inherits this component. Also, shouldn't this be an array of strings? Or is Rails magically converting new lines into spaces?

Reply:
Thanks for your feedback, @tvdeyen. I agree. I changed it to be defined in a method and as an array.